engine/verification/requester/requester.go

```diff
@@ -248,12 +251,18 @@ func (e *Engine) onTimer() {
 	}
 
 	for _, request := range pendingReqs {
-		e.handleChunkDataPackRequestWithTracing(request, lastSealed.Height)
+		attempts := e.handleChunkDataPackRequestWithTracing(request, lastSealed.Height)
+		if attempts > maxAttempts {
+			maxAttempts = attempts
+		}
```

Comment:
instead of checking the max attempts of _any_ chunks, what about just keep track of the number of attempts for the next unsealed chunk data pack?

We have the lastSealed.Height, so just need to find a pending req to a chunk, which has height equals to lastSealed.Height + 1.

We probably will only care about this metrics in the event of sealing halt, in which case, we care the most about the next unsealed block, a metrics focused on the next unsealed block would be more useful than a max attempts to a chunk which might not be the next unsealed block (for instance, a malicious result that we will ever receive chunk data pack)

Reply:
@zhangchiqing implemented in https://github.com/onflow/flow-go/pull/1290/commits/597fcd91de37e9cd1c4944d6fb3a5296978552ef, https://github.com/onflow/flow-go/pull/1290/commits/1c369f845e8fc893ebc65bda29bbfcb75e8ccc95, and https://github.com/onflow/flow-go/commit/0b7a2ced5b61dea0be3379f7e8803ba9a087028f